### PR TITLE
created models for races and enums

### PR DESCRIPTION
**Pull request by LukeAVanDrie**
WIP: I promise to never make a PR this big again (only if you promise to never make a ticket this big again)

### CODE REVIEW

File: src/components/characterSheet.js [hunk 1]
```diff
@@ -3,15 +3,25 @@ import { Redirect } from "react-router-dom";
 import Layout from '@wedgekit/layout';
 import { Button, Card, StackedCard } from '@wedgekit/core';
 
+const styles = {
```
Comment (sarahdmsi) [inline src/components/characterSheet.js:6]: I'd prefer that you used `styled-components` for cases like this. Would look something like:

```
const StyledCard = styled(Card)`
  margin: 3em 3em 0em 3em;
`
```

File: src/components/characterForm.js [hunk 2]
```diff
@@ -1,57 +1,332 @@
...
+            languageDropdown: false,
+            toolProficiencyDropdown: false,
+            ancestryDropdown: false,
+
+            // character properties
+            character: character,
+            race: character ? character.race : undefined,
+
+            // form data
+            formName: formData ? formData.formName : "",
```
Comment (sarahdmsi) [inline src/components/characterForm.js:58]: Something like `dot-prop` might help you get rid of a lot of ternary checks.

File: src/components/characterForm.js [hunk 3]
```diff
@@ -1,57 +1,332 @@
...
-                            ?  JSON.parse(localStorage.getItem("values")).name
-                            : "" 
+        const renderSubraceDropdown = () => {
+            if (this.state.subraceDropdown) {
+                return (
+                    <Field
+                        name="subRace"
+                        label="Subrace"
+                    >
+                        {({ fieldProps }) => <Dropdown { ...fieldProps }
```
Comment (sarahdmsi) [inline src/components/characterForm.js:179]: I feel like the `Select` component would be preferable here than the dropdown, unless I'm misunderstanding your intent. Additionally, it has a cleaner API for use in forms.

File: src/components/characterForm.js [hunk 4]
```diff
@@ -1,57 +1,332 @@
...
+    Dragonborn, HillDwarf, MountainDwarf,
+    HighElf, WoodElf, DarkElf,
+    ForestGnome, RockGnome, HalfElf,
+    LightfootHalfling, StoutHalfling, HalfOrc,
+    Human, Tiefling, Race 
+} from "../models/races/bundle";
+import { LanguageEnum, ToolEnum, AncestryEnum, SizeEnum } from "../models/enums/bundle"; 
+
+const styles = {
+    card: {
```
Comment (sarahdmsi) [inline src/components/characterForm.js:16]: I added a comment on another file about using `styled-components`, but this is another good case for it.

File: src/components/characterForm.js [hunk 5]
```diff
@@ -1,57 +1,332 @@
...
+                    languageDropdown: result.requiredArgs.includes("language"),
+                    toolProficiencyDropdown: result.requiredArgs.includes("toolProficiency"),
+                    ancestryDropdown: result.requiredArgs.includes("ancestry")
+                });
+            }
+        });
+    }
+
+    componentDidMount() {
+        this.generateRace();
```
Comment (sarahdmsi) [inline src/components/characterForm.js:164]: We largely have moved away from `componentDidMount` methods in Agility by using memoize in render.

File: src/App.js [hunk 6]
```diff
@@ -9,6 +9,8 @@ import CharacterForm from "./components/characterForm";
 import CharacterSheet from "./components/characterSheet";
 
 export default function App() {  
+  document.body.style = "background: #1B2027";
```
[the remark this answers is not in the document]
Comment (LukeAVanDrie) [reply src/App.js]: resolved

**Pull request merged**